lucene/benchmark/src/java/org/apache/lucene/benchmark/byTask/tasks/SearchWithCollectorTask.java

```
@@ -46,17 +46,17 @@ public boolean withCollector() {
   }
 
   @Override
-  protected Collector createCollector() throws Exception {
-    Collector collector = null;
+  protected CollectorManager<?, ?> createCollectorManager() throws Exception {
+    CollectorManager<?, ?> collectorManager;
     if (clnName.equalsIgnoreCase("topScoreDoc") == true) {
-      collector = TopScoreDocCollector.create(numHits(), Integer.MAX_VALUE);
+      collectorManager = new TopScoreDocCollectorManager(numHits(), Integer.MAX_VALUE);
     } else if (clnName.length() > 0) {
-      collector = Class.forName(clnName).asSubclass(Collector.class).getConstructor().newInstance();
-
+      collectorManager =
+          Class.forName(clnName).asSubclass(CollectorManager.class).getConstructor().newInstance();
     } else {
-      collector = super.createCollector();
+      collectorManager = super.createCollectorManager();
     }
-    return collector;
+    return collectorManager;
```

I am not familiar with this utility and I wonder if this type of change is acceptable for a major release, whether there's other ways. If we continue down this route, how do we document the breaking change, and should we deprecate createCollector in the 9x branch?

---

I have now added a changes entry, and replaced collector.class with a new config parameter. We throw error once collector.class is provided, so we fail fast and users know what they have to do.